### PR TITLE
Add "No New Privileges" restriction and capability restrictions, as well as AppArmor and seccomp profiles

### PR DESCRIPTION
**Pull request by ChrysoliteAzalea**
Hello everyone! I propose enabling a "No New Privileges" restriction and capabilities restrictions to the Waydroid container, as well as adding seccomp and AppArmor profiles.

"No New Privileges" is a restrictive flag on a process that prevents a process from changing its effective user or group ID or its capabilities by executing a new process, also this restricts AppArmor profile transitions to subsets of a current profile.

Also, I propose adding a capability restrictions to the container. I've found what capabilities are needed by the container by creating an AppArmor profile for the container, which I have proposed in [an issue 401](https://github.com/waydroid/waydroid/issues/401) and this pull request

AppArmor is a mandatory access control system for GNU/Linux that restricts process's access to files, sockets, privileged operations etc. I've created three AppArmor profiles for this container (one generic, one for ADBD, one for apps).

Seccomp is a Linux kernel feature that restricts system calls that can be made by a process. I've added a seccomp profile for a container that blocks several potentially dangerous syscalls (for example, rebooting or loading a kernel module).

### CODE REVIEW

**Comment by Quackdoc** [comment]
needs added to config_1 as well

**Comment by ChrysoliteAzalea** [comment]
The previous config was causing **logcat read failure** on my system -- had to grant CAP_AUDIT_CONTROL to container in order to fix it. Also, removed the universal read access from the **android_app** profile, replacing it with more fine-grained rules.

File: tools/helpers/lxc.py [hunk 1]
```diff
@@ -128,16 +128,29 @@ def set_lxc_config(args):
...
     command = ["cp", "-fpr", config_path, lxc_path + "/config"]
     tools.helpers.run.user(args, command)
+    command = ["cp", "-fpr", seccomp_profile, lxc_path + "/waydroid.seccomp"]
+    tools.helpers.run.user(args, command)
+    command = ["cp", "-fpr", apparmor_profiles[0], apparmor_profile_dir + "lxc/lxc-waydroid"]
+    tools.helpers.run.user(args, command)
+    command = ["cp", "-fpr", apparmor_profiles[1], apparmor_profile_dir + "android_app"]
+    tools.helpers.run.user(args, command)
+    command = ["cp", "-fpr", apparmor_profiles[2], apparmor_profile_dir + "adbd"]
+    tools.helpers.run.user(args, command)
```
[the remark this answers is not in the document]
Comment (ChrysoliteAzalea) [reply tools/helpers/lxc.py]: OK, added a try-except block with a warning if installation of profiles fails.

**Comment by aleasto** [comment]
See also: https://github.com/waydroid/android_vendor_waydroid/pull/11

**Comment by ChrysoliteAzalea** [comment]
Force-pushed the branch to merge commits into one and sign it. No files have been changed.

**Comment by aleasto** [comment]
Any tests happening with this? I'm not comfortable blindly merging it

**Comment by ChrysoliteAzalea** [comment]
I've tested it on my system, and it seems working fine. However, there are several use-cases I'm not sure about:

* Systems that don't use AppArmor. I have added a correction that warns a user in case that copying AppArmor profiles has failed, but I haven't tested it (I think that disabling AppArmor would be a security risk since I have a lot of software sandboxed with AppArmor)
* I haven't tested root solutions such as Magisk, and I don't know how container-wide NNP affects root, and since with NNP there is no way to transition to a less-restricted domain, an app running as root may still be affected by AppArmor policies (I haven't tested Magisk because I don't want to get root in a privileged container)
* I also haven't tested the Google Play Services, and don't know how they are affected by AppArmor and other restrictions (I don't need Google Play Services)

So, I think that we could close this pull request and split it into four smaller parts:

* No New Privileges
* AppArmor
* seccomp
* capabilities

I think that we can restrict capabilities, since it would only affect apps running as root (and on non-rooted containers, it would only affect system components, and I have tested it on my container). I think we can also add LXC-level seccomp (however I don't know whether it's necessary, since apps are already more tightly confined with a zygote-level seccomp). NNP should be tested by rooted container users and by Google Play Services container users before merging, and AppArmor profiles should be loaded in "complain" mode first.

**Comment by ChrysoliteAzalea** [comment]
Closing this in favor of #503, #504, #505 and #506.

**Comment by warlice** [comment]
It's a bad ieda to add secmop noNewprivs. because, the nomral android enviroment don't have these limits. so waydroid was detected as emulater

**Pull request closed without merging**